### PR TITLE
Web Actions Tab 0.1.3 - 0.1.4

### PR DESCRIPTION
**Pull request by adrw**
* Move client side processing of metadata to Redux so it is not render blocking
* Skeleton Loading Component instead of using `<Spinner/>`

### CODE REVIEW

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx [hunk 1]
```diff
@@ -288,81 +256,64 @@ export const FilterWebActions = (props: IState & IDispatchProps) => {
     </div>
   )
 }
+const SkeletonText = () => (
+  <span className={Classes.SKELETON}>{"Lorem ipsum"}</span>
```
Comment (wesleyk) [inline misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: hmm will this ever show up to devs?

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx [hunk 2]
```diff
@@ -289,80 +257,18 @@ export const FilterWebActions = (props: IState & IDispatchProps) => {
...
-      {/* <FilterWebActions {...props} /> */}
-    </div>
-  )
+const WebActionsContainer = (props: IState & IDispatchProps) => {
+  const metadata = simpleSelect(props.webActions, "metadata")
+  if (metadata.length > 0) {
+    return (
+      <div>
+        <WebActions metadata={metadata} {...props} />
+        {/* <FilterWebActions {...props} /> */}
```
Comment (wesleyk) [inline misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: this is confusing, is this commented out?

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx (hunk 1 above)
Comment (adrw) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: If computation takes too long, usually may flash prior to being filled in with the real data. Similar to the Skeleton table we already use in SampleTableComponent in Misk-Web/Palette

https://square.github.io/misk-web/examples/tabs/palette/demo/

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx (hunk 2 above)
Comment (adrw) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: Yes commenting with `JSX` is a bit confusing.

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx (hunk 1 above)
Comment (wesleyk) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: why not put no txt?

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx (hunk 2 above)
Comment (wesleyk) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: so this is a comment, why are we leaving this here?

Comment (adrw) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: Will remove 👍

File: misk/web/tabs/web-actions/src/ducks/webActions.ts [hunk 3]
```diff
@@ -90,8 +134,82 @@ function* handleDinosaur(action: IAction<WEBACTIONS, IWebActionsPayload>) {
...
+          action => action.dispatchMechanism
+        )
+        const mergedAction = actions[0]
+        mergedAction.dispatchMechanism = dispatchMechanism.sort().reverse()
+        return mergedAction
+      })
+      .sortBy(["name", "pathPattern"])
+      .value()
+    // TODO(adrw) build index of keyspace for filterable fields
+    // const index = chain(metadata).value()
```
Comment (wesleyk) [inline misk/web/tabs/web-actions/src/ducks/webActions.ts]: same comment about not committing commented out code like this

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx (hunk 1 above)
Comment (adrw) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: A bit unintuitive at first glance but allows a way to mock the final UI when the data has loaded: https://blueprintjs.com/docs/#core/components/skeleton

**Pull request merged**